### PR TITLE
Enforce check-in, check-out ids to be same as the tx timestamp

### PR DESCRIPTION
**Pull request by liayoo**
To make sure they're unique, only allow the check-in/check-out ids that are the same as the tx timestamp.

### CODE REVIEW

File: db/rule-util.js [hunk 1]
```diff
@@ -272,14 +273,15 @@ class RuleUtil {
     if (!_.isEqual(request, newData.request)) {
       return false;
     }
-    return this.isDict(newData.response) && this.isValidHash(newData.response.tx_hash) &&
-        this.isBool(newData.response.status);
+    return this.isDict(newData.response) && this.isBool(newData.response.status) &&
+        (newData.response.tx_hash === undefined || this.isValidHash(newData.response.tx_hash));
```
Comment (platfowner) [inline db/rule-util.js:277]: minor: would it be helpful for the readers if some comments about the undefined case is available?

File: db/rule-util.js [hunk 2]
```diff
@@ -242,11 +242,12 @@ class RuleUtil {
...
   }
 
-  validateCheckoutRequestData(networkName, chainId, tokenId, userAddr, checkoutId, newData, getValue) {
+  validateCheckoutRequestData(networkName, chainId, tokenId, userAddr, checkoutId, newData, currentTime, getValue) {
     const { PredefinedDbPaths } = require('../common/constants');
     const PathUtil = require('../common/path-util');
     if (!this.isDict(newData) || !this.isNumber(newData.amount) || newData.amount <= 0 ||
-        !this.isString(newData.recipient) || !this.isNumber(newData.fee_rate)) {
+        !this.isString(newData.recipient) || !this.isNumber(newData.fee_rate) ||
+        Number(checkoutId) !== currentTime) {
```
Comment (platfowner) [inline db/rule-util.js:250]: would it be helpful if some comments are available about the restriction on the checkoutId?

File: db/rule-util.js [hunk 3]
```diff
@@ -272,14 +273,15 @@ class RuleUtil {
...
+        (newData.response.tx_hash === undefined || this.isValidHash(newData.response.tx_hash));
   }
 
-  validateCheckinRequestData(networkName, chainId, tokenId, userAddr, checkinId, newData, getValue) {
+  validateCheckinRequestData(networkName, chainId, tokenId, userAddr, checkinId, newData, currentTime, getValue) {
     const PathUtil = require('../common/path-util');
     if (!this.isDict(newData) || !this.isNumber(newData.amount) || newData.amount <= 0 ||
-        !this.isString(newData.sender) || !this.isString(newData.sender_proof)) {
+        !this.isString(newData.sender) || !this.isString(newData.sender_proof) ||
+        Number(checkinId) !== currentTime) {
```
Comment (platfowner) [inline db/rule-util.js:284]: ditto

**Review by platfowner — APPROVED** [review]
LGTM with some comments.
Thanks!

File: db/rule-util.js (hunk 1 above)
Comment (liayoo) [reply db/rule-util.js:279]: Right, done!

File: db/rule-util.js (hunk 2 above)
Comment (liayoo) [reply db/rule-util.js:251]: Done!

File: db/rule-util.js (hunk 3 above)
Comment (liayoo) [reply db/rule-util.js:287]: ditto

**Pull request merged**